library/src/main/java/io/ona/kujaku/location/clients/AndroidGpsLocationClient.java

```diff
@@ -114,10 +124,12 @@ public void onResult(@NonNull LocationSettingsResult locationSettingsResult) {
         }
     }
 
+    @SuppressLint("MissingPermission")
     @Nullable
     @Override
     public Location getLastLocation() {
-        return lastLocation;
+        Location lastLocationFromProvider = locationManager.getLastKnownLocation(getProvider());
+        return lastLocationFromProvider != null ? lastLocationFromProvider : lastLocation;
```

Comment:
can we compare location age between lastLocationFromProvider and lastLocation and return the newest

Reply:
Cool